src/picterra/client.py

```diff
@@ -238,6 +238,43 @@ def create_detector(self, name: str = '', type: str = 'count') -> str:
             raise APIError(resp.text)
         return resp.json()['id']
 
+    def list_detectors(self):
+        """
+        Returns:
+            A list of detectors dictionaries
+
+        Example:
+
+            ::
+
+                {
+                    'id': '42',
+                    'detection_type': 'count',
+                    'name': 'cow detector',
+                    'output_type': 'bbox',
+                    'training_steps': 787
+                },
+                {
+                    'id': '43',
+                    'detection_type': 'segmentation',
+                    'name': 'test5',
+                    'output_type': 'polygon',
+                    'training_steps': 500
+                }
+
+        """
+        url = self._api_url('detectors/?page_number=1')
+        data = []
+        while url:
+            logger.debug('Paginating through detectors list at page %s' % url)
+            resp = self.sess.get(url)
+            r = resp.json()
+            url = r['next']
+            count = r['count']
+            data += r['results']
```

Comment:
Is this something we could extract in a helper function ? Seems like we already have the same logic for rasters listing.

Reply:
Checked